File: CoWorkHub/CoWorkHub.Model/SpaceUnitImage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class SpaceUnitImage
    {
        public int ImageId { get; set; }
        public int SpaceUnitId { get; set; }
        public string ImagePath { get; set; } = null!;
        public bool IsDeleted { get; set; } = false;
    }
}

File: CoWorkHub/UI/coworkhub_mobile/lib/models/space_unit_image.dart
import 'package:json_annotation/json_annotation.dart';

part 'space_unit_image.g.dart';

@JsonSerializable()
class SpaceUnitImage {
  int imageId;
  int spaceUnitId;
  String imagePath;
  bool isDeleted;

  SpaceUnitImage({
    required this.imageId,
    required this.spaceUnitId,
    required this.imagePath,
    this.isDeleted = false,
  });

  factory SpaceUnitImage.fromJson(Map<String, dynamic> json) =>
      _$SpaceUnitImageFromJson(json);

  Map<String, dynamic> toJson() => _$SpaceUnitImageToJson(this);
}
